Remove hipMathFunctions.cc test which requires -fallow-half-arguments-and-returns
See https://github.com/ROCm-Developer-Tools/HIP/issues/3178
===========================================================
Index: HIP-rocm-5.7.1/tests/catch/unit/deviceLib/CMakeLists.txt
===================================================================
--- HIP-rocm-5.7.1.orig/tests/catch/unit/deviceLib/CMakeLists.txt
+++ HIP-rocm-5.7.1/tests/catch/unit/deviceLib/CMakeLists.txt
@@ -36,7 +36,6 @@ set(AMD_TEST_SRC
     bitExtract.cc
     bitInsert.cc
     floatTM.cc
-    hipMathFunctions.cc
     hmax_hmin.cc
     bfloat16.cc
 )
@@ -121,7 +120,6 @@ if(${ARCH_CHECK} GREATER_EQUAL 0)
     set_source_files_properties(unsafeAtomicAdd_NonCoherent_withunsafeflag.cc PROPERTIES COMPILE_OPTIONS "-munsafe-fp-atomics")
     set_source_files_properties(unsafeAtomicAdd_Coherent_withnounsafeflag.cc PROPERTIES COMPILE_OPTIONS "-mno-unsafe-fp-atomics")
     set_source_files_properties(unsafeAtomicAdd_NonCoherent_withnounsafeflag.cc PROPERTIES COMPILE_OPTIONS "-mno-unsafe-fp-atomics")
-    set_source_files_properties(hipMathFunctions.cc PROPERTIES COMPILE_FLAGS "-Xclang -fallow-half-arguments-and-returns")
     file(GLOB AtomicAdd_files *AtomicAdd_*_*.cc)
     set_property(SOURCE ${AtomicAdd_files} PROPERTY COMPILE_FLAGS --save-temps)
     file(GLOB unsafeAtomicAdd_files *unsafeAtomicAdd_*_*.cc)
